move/interchain_token_service/sources/types/token_id.move

```diff
@@ -62,7 +65,9 @@ module interchain_token_service::token_id {
     }
 
     public fun unregistered_token_id(symbol: &ascii::String, decimals: u8): UnregisteredTokenId {
-        let mut v = vector[decimals];
+        let prefix = PREFIX_UNREGISTERED_INTERHCAIN_TOKEN_ID;
+        let mut v = bcs::to_bytes(&prefix);
+        v.push_back(decimals);
         v.append(*ascii::as_bytes(symbol));
         let id = address::from_bytes(keccak256(&v));
```

Comment:
Add tests to ensure any critical cryptographic derivation doesn't change unexpectedly (similar to how we check for EVM ITS)